toolkit.nu

```diff
@@ -211,3 +211,41 @@ export def "check pr" [
 
     report --no-fail
 }
+
+# creates git hooks to run:
+# - `toolkit fmt` on `git commit`
+# - `toolkit fmt --check` and `toolkit clippy` on `git push`
+export def set-git-hooks [] {
+    set-git-hook pre-commit `#!/usr/bin/env nu
+
+use ../../toolkit.nu [fmt, pretty-print-command]
+
+print $"running ('toolkit fmt' | pretty-print-command)"
+fmt`
```

Comment:
here i think it should just fail :thinking: 

if you have bad formatting, `toolkit fmt` would fix that but not stage the format changes to be commited.

you could try to extract back the formatting patch given by `cargo fmt` here and apply it, but
- that might be tideous 
- personally, i do not like automatic commits / patching :yum:

Reply:
Ahh you're right, I did notice the formatting happen but the changes still being in my local tree, but I thought this was something funky with my editor's source control pane.

And I agree, I think the commit should just fail and let the user fix the formatting without doing it behind the scenes :)